core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -89,9 +89,15 @@ private[spark] abstract class Task[T](
     } finally {
       context.markTaskCompleted()
       try {
+        val shuffleMemoryManager = SparkEnv.get.shuffleMemoryManager
+        val shuffleMemoryUsed = shuffleMemoryManager.getMemoryConsumptionForThisTask()
+        if (SparkEnv.get.conf.contains("spark.testing") && shuffleMemoryUsed != 0) {
+          throw new SparkException(
```

Comment:
In case of failed task, it's fine to have some memory leak (will be freed finally), will this exception override the real one?